src/client/render/core.cpp

```diff
@@ -67,6 +86,87 @@ void RenderingCore::draw(video::SColor _skycolor, bool _show_hud, bool _show_min
 	draw_crosshair = _draw_crosshair;
 
 	beforeDraw();
+
+	// todo - use a simpler shader for shadow map generation
+	static u32 shadow_shader;
+	if (shadow_shader == 0) {
+		IShaderSource *shdsrc = client->getShaderSource();
+		shadow_shader = shdsrc->getShader("shadow_shader", TILE_MATERIAL_OPAQUE, 0);
+	}
+
+	client->getEnv().getClientMap().depthTexture = NULL;
+
+	{
+		static COpenGLRenderTarget *lightScene;
+		if (lightScene == NULL) {
+#ifdef _IRR_WINDOWS_API_
+			glActiveTexture = (PFNGLACTIVETEXTUREPROC)IRR_OGL_LOAD_EXTENSION("glActiveTexture");
+#endif
+			// ARB FrameBufferObjects
+			glBindFramebuffer = (PFNGLBINDFRAMEBUFFERPROC)IRR_OGL_LOAD_EXTENSION("glBindFramebuffer");
+			glDeleteFramebuffers = (PFNGLDELETEFRAMEBUFFERSPROC)IRR_OGL_LOAD_EXTENSION("glDeleteFramebuffers");
+			glGenFramebuffers = (PFNGLGENFRAMEBUFFERSPROC)IRR_OGL_LOAD_EXTENSION("glGenFramebuffers");
+			glCheckFramebufferStatus = (PFNGLCHECKFRAMEBUFFERSTATUSPROC)IRR_OGL_LOAD_EXTENSION("glCheckFramebufferStatus");
+			glFramebufferTexture2D = (PFNGLFRAMEBUFFERTEXTURE2DPROC)IRR_OGL_LOAD_EXTENSION("glFramebufferTexture2D");
+
+			u32 SHADOWMAP_RES = 2048;
+			lightScene = new COpenGLRenderTarget(v2u32(SHADOWMAP_RES, SHADOWMAP_RES), "lightscene", true, driver);
+		}
+		//driver->setRenderTarget(0, true, true, video::SColor(0xFFFFFFFF));
+
+		// Technically, we don't need a color buffer for standard shadow mapping
+		lightScene->bindRTT(false, true);
+
+		sky.OnAnimate(porting::getTimeMs());
```

Comment:
Do we have to call `OnAnimate()` to render things correctly? Isn't this already done in the normal draw?
Can this be source of glitches (faster animation, bad timing, etc.)?